### PR TITLE
Update with Network Override for iOS

### PR DESCRIPTION
**Pull request by cdhoffmann**

### CODE REVIEW

File: using-mobile-extensions/mobile-core/platform-services.md [hunk 1]
```diff
@@ -146,3 +148,130 @@ This step should occur prior to any other interactions with the AEP SDK. While i
     }
 ```
 
+## iOS
+
+### 1. Conform to the ACPHttpConnectionPerformer protocol
+
+The ACPHttpConnectionPerformer is a protocol which must be conformed to in order to override the network stack. It provides two methods which must be implemented:
+
+    1. `shouldOverride`, which takes a URL and an Http method, and returns true if the network stack should be overriden by     the performer, or false if it should not.
```
Comment (shalehaha) [inline using-mobile-extensions/mobile-core/platform-services.md:157]: HTTP

File: using-mobile-extensions/mobile-core/platform-services.md [hunk 2]
```diff
@@ -146,3 +148,130 @@ This step should occur prior to any other interactions with the AEP SDK. While i
...
+### 1. Conform to the ACPHttpConnectionPerformer protocol
+
+The ACPHttpConnectionPerformer is a protocol which must be conformed to in order to override the network stack. It provides two methods which must be implemented:
+
+    1. `shouldOverride`, which takes a URL and an Http method, and returns true if the network stack should be overriden by     the performer, or false if it should not. 
+    2. `requestUrl`, which is the URL request override with a completion block.
+    
+### 2. Create an ACPHttpConnection and pass relevant data from your NSURLSessionDataTask completion block
+
+The completion block for the `requestUrl` method, takes an `ACPHttpConnection` as it's parameter. The ACPHttpConnection is used when overriding the network stack in place of the internal network connection implementation and represents the response to an Http request. It is to be created using the NSURLResponse, and NSData from your url request response. In the case of a Network Error, or timeout, the `ACPHttpConnection*` is expected to be nil.
```
Comment (shalehaha) [inline using-mobile-extensions/mobile-core/platform-services.md:162]: it's -> its

Comment (shalehaha) [reply using-mobile-extensions/mobile-core/platform-services.md:162]: ACPHttpConnection -> `ACPHttpConnection `

Comment (shalehaha) [reply using-mobile-extensions/mobile-core/platform-services.md:162]: http -> HTTP

File: using-mobile-extensions/mobile-core/platform-services.md (hunk 1 above)
Comment (shalehaha) [reply using-mobile-extensions/mobile-core/platform-services.md:157]: should make it clear that they decide whether to return true of false

File: using-mobile-extensions/mobile-core/platform-services.md [hunk 3]
```diff
@@ -146,3 +148,130 @@ This step should occur prior to any other interactions with the AEP SDK. While i
...
+                                          NSURLResponse * response,
+            NSError * error) {
+                if(!error) {
+                    NSHTTPURLResponse* httpResponse = (NSHTTPURLResponse*)response;
+                    
+            // ***** NOTE the creation of the ACPHttpConnection outlined in step 2.*****
+            ACPHttpConnection* connOverride = [[ACPHttpConnection alloc] initWithResponse:httpResponse data:data];
+            completion(connOverride);
+        } else if ([[error domain]  isEqual:@"NSURLErrorDomain"]) {
+            completion(NULL);
```
Comment (shalehaha) [inline using-mobile-extensions/mobile-core/platform-services.md:203]: nil

File: using-mobile-extensions/mobile-core/platform-services.md [hunk 4]
```diff
@@ -146,3 +148,130 @@ This step should occur prior to any other interactions with the AEP SDK. While i
...
+        } else {
+            completion(NULL);
```
Comment (shalehaha) [inline using-mobile-extensions/mobile-core/platform-services.md:205]: nil

File: using-mobile-extensions/mobile-core/platform-services.md (hunk 3 above)
Comment (shalehaha) [reply using-mobile-extensions/mobile-core/platform-services.md:203]: looks like we don't need this `else if` block

File: using-mobile-extensions/mobile-core/platform-services.md [hunk 5]
```diff
@@ -146,3 +148,130 @@ This step should occur prior to any other interactions with the AEP SDK. While i
...
+
+```objectivec
+@implementation AppDelegate
+
+-(BOOL)application:(UIApplication *)application 
+didFinishLaunchingWithOptions:(NSDictionary<UIApplicationLaunchOptionsKey, id> *)launchOptions {
+        ...
+        [ACPNetworkServiceOverrider setHttpConnectionPerformer:[[YourPerformerOverrider alloc] init]];
+        ...
+        ACPCore.start {
```
Comment (shalehaha) [inline using-mobile-extensions/mobile-core/platform-services.md:259]: this is swift

File: using-mobile-extensions/mobile-core/platform-services.md [hunk 6]
```diff
@@ -7,9 +7,11 @@ Currently, the AEP SDK only provides the capability for overriding the Adobe-pro
 This section walks through the steps necessary to create a custom network override, and register it with the SDK.
 
 {% hint style="info" %}
-This feature is only available in Android Core version 2.5.0 or later.
+This feature is only available in Android Core version 2.5.0 or later, and iOS Core version 2.6.0 or later.
```
Comment (ahmyr) [inline using-mobile-extensions/mobile-core/platform-services.md:10]: Take out the comma in this sentence.

File: using-mobile-extensions/mobile-core/platform-services.md [hunk 7]
```diff
@@ -146,3 +148,128 @@ This step should occur prior to any other interactions with the AEP SDK. While i
     }
 ```
 
+## iOS
+
+### 1. Conform to the ACPHttpConnectionPerformer protocol
+
+The `ACPHttpConnectionPerformer` is a protocol which must be conformed to in order to override the network stack. It provides two methods which must be implemented:
+
+    1. `shouldOverride`, which takes a URL and an HTTP method. You return true if you want the network stack to be overriden     by the performer, or false if not.
```
Comment (ahmyr) [inline using-mobile-extensions/mobile-core/platform-services.md:157]: Take out the "You". Should be "overridden" (with two ds). Also, looks like there is some whitespace between "overridden" and "by"

File: using-mobile-extensions/mobile-core/platform-services.md [hunk 8]
```diff
@@ -146,3 +148,128 @@ This step should occur prior to any other interactions with the AEP SDK. While i
...
+### 1. Conform to the ACPHttpConnectionPerformer protocol
+
+The `ACPHttpConnectionPerformer` is a protocol which must be conformed to in order to override the network stack. It provides two methods which must be implemented:
+
+    1. `shouldOverride`, which takes a URL and an HTTP method. You return true if you want the network stack to be overriden     by the performer, or false if not. 
+    2. `requestUrl`, which is the URL request override with a completion block.
+    
+### 2. Create an ACPHttpConnection and pass relevant data from your NSURLSessionDataTask completion block
+
+The completion block for the `requestUrl` method, takes an `ACPHttpConnection` as its parameter. The `ACPHttpConnection` is used when overriding the network stack in place of the internal network connection implementation and represents the response to an HTTP request. It is to be created using the NSURLResponse, and NSData from your url request response. In the case of a Network Error, or timeout, the `ACPHttpConnection*` is expected to be nil.
```
Comment (ahmyr) [inline using-mobile-extensions/mobile-core/platform-services.md:162]: Take the comma out of this sentence: The completion block for the `requestUrl` method, takes an `ACPHttpConnection` as its parameter.

Comment (ahmyr) [reply using-mobile-extensions/mobile-core/platform-services.md:162]: Take the comma out of this sentence: It is to be created using the NSURLResponse, and NSData from your url request response.

**Pull request merged**